model-optimizer/extensions/ops/loop.py

```diff
@@ -61,6 +61,27 @@ def get_body_node_by_internal_id(loop_node: Node, internal_id: int):
             'Expected 0 or 1 node with `internal_layer_id`={}, {} found'.format(internal_id, len(suitable_nodes))
         return suitable_nodes[0] if len(suitable_nodes) == 1 else None
 
+    @staticmethod
+    def get_external_nodes_by_internal_id(loop_node: Node, internal_layer_id: int) -> list:
+        """
+        Get a list of nodes from the main graph that are connected with a node with internal_layer_id
+        from the body graph
+
+        :param loop_node: The Loop node
+        :param internal_layer_id: Internal layer ID of the node in the body graph
+        :return: A list of external nodes (from the main graph) that are connected with a node with
+        internal_layer_id from the body graph
+        """
+        for map_item in loop_node.input_port_map:
+            if map_item['internal_layer_id'] == internal_layer_id \
+                    and not loop_node.in_port(map_item['external_port_id']).disconnected():
```

Comment:
nit: change `not loop_node.in_port(map_item['external_port_id']).disconnected():` to `loop_node.is_in_port_connected(map_item['external_port_id'])`. The proposed solution will not fail if the port with number `map_item['external_port_id']` does not exist in the node. The same comment for line 81.

Reply:
Fixed. Thanks.